[Fix] Validate items for saleable, purchaseable or subcontractable in transactions

Zlash65 opened the pull request:
![validate-items](https://user-images.githubusercontent.com/11695402/40840717-7b49e130-65c5-11e8-8e7d-ea851a8f6838.gif)

Summary:- Items that have `is_sales_item` unchecked are not shown in dropdown but can be manually entered and transactions can be created against it. Same goes for `is_purchase_item` & `is_subcontract_item`. Validation for such provision added.

Zlash65 committed:
validate items to see if they are saleable

Zlash65 committed:
check if items are subcontractable or saleable

nabinhait commented on erpnext/controllers/buying_controller.py:
```diff
@@ -456,3 +457,21 @@ def validate_schedule_date(self):
+		""".format(item_condition, condition), as_list=True)]
+
+	if invalid_items:
+		frappe.throw(_("{0} items: {1}".format(message, ", ".join([d for d in invalid_items]))))
```
Improve the messages. "Non subcontractable items" or "Non purchaseable items" are not that good.

Zlash65 committed:
improvise name,error message and code
Validate type function validates item if they are of proper type for that transaction and throws descriptive error.

Zlash65 commented:
<img width="601" alt="screen shot 2018-06-04 at 11 45 16 am" src="https://user-images.githubusercontent.com/11695402/40901416-c2011902-67ed-11e8-9f62-b97407413b20.png">
